core/modules/layout/includes/layout.class.inc

```diff
@@ -831,14 +836,23 @@ class Layout {
       ));
     }
 
+    $contexts_cache[$this->name] = $this->contexts;
+
+    return $this->getContextsByType($this->contexts, $include_types);
+  }
+
+  /**
+   * Get context by requested type.
+   */
+  function getContextsByType($contexts, $include_types = LayoutContext::USAGE_TYPE_ALL) {
```

Comment:
Some coding standard improvement could be:

- declare visibility on the new function
- document the parameters and return

Reply:
> document the parameters and return

This is true, will do
> declare visibility on the new function

I dont understand this one though.

Reply:
Re visibility: it could be protected of public in the Layout class, whatever fits best. If not declared, it defaults to "public", but it's usually best practice to actually declare it.

Reply:
Ahh, visibility, I get it now. Hadnt realised this was what this was called. Will add. I think protected, since it wouldnt really make sense to be accessible publicly.

Reply:
Yeah, I didn't know what it was called either - I always referred to it as "scope" 😅

Reply:
Done